user-impl/src/main/java/com/example/auction/user/impl/UserRepository.java

```diff
@@ -0,0 +1,199 @@
+        private final CassandraReadSide readSide;
+
+        private PreparedStatement insertUserStatement;
+
+        @Inject
+        public PUserEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<PUserEvent> buildHandler() {
+            return readSide.<PUserEvent>builder("pUserEventOffset")
+                    .setGlobalPrepare(this::createTables)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(PUserEvent.PUserCreated.class,
+                            e -> insertUser(e.getUser()))
+
+                    .build();
+        }
+
+        private void registerCodec(Session session, InstantCodec codec) {
+            session.getCluster().getConfiguration().getCodecRegistry().register(codec);
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<PUserEvent>> aggregateTags() {
+            return PUserEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTables() {
+            return doAll(
+                    session.executeCreateTable(
+                            "CREATE TABLE IF NOT EXISTS UserInfo (" +
+                                    "UserId UUID, " +
+                                    "createdAt timestamp, " +
+                                    "Name text, " +
+                                    "email text, " +
+                                    "PRIMARY KEY (userId, createdAt) " +
+                                    ")" +
+                                    "WITH CLUSTERING ORDER BY (createdAt DESC)"
+                    )
+
+            );
+        }
+
+        private CompletionStage<Done> prepareStatements() {
+            return doAll(
+                    session.underlying()
+                            .thenAccept(s -> registerCodec(s, InstantCodec.instance))
+                            .thenApply(x -> Done.getInstance()),
+                    prepareInsertUserStatement());
+
+
+        }
+
+
+        private CompletionStage<Done> prepareInsertUserStatement() {
+
+            return session.
+                    prepare("INSERT INTO UserInfo(" +
+
+                            "UserId, " +
+                            "createdAt, " +
+                            "Name, " +
+                            "email" +
+
+                            ") VALUES (" +
+                            "?, " + // UserId
+                            "?, " + // createdAt
+                            "?, " + // Name
+                            "?" + // email
+
+                            ")"
+                    )
+                    .thenApply(accept(s -> insertUserStatement = s));
+        }
+
+
+        private CompletionStage<List<BoundStatement>> insertUser(PUser user) {
+            return completedStatements(
+                    Arrays.asList(insertUserCreator(user))
```

Comment:
Here we can remove the `Arrays.asList` because there is an overloaded `completedStatements` that gets a varargs.